envoy/http/filter.h

```diff
@@ -1079,14 +1079,16 @@ class FilterChainFactoryCallbacks {
 using FilterFactoryCb = std::function<void(FilterChainFactoryCallbacks& callbacks)>;
 
 /**
- * Simple struct of additional contextual information of filter factory, e.g. custom filter name
+ * Simple struct of additional contextual information of HTTP filter, e.g. filter config name
  * from configuration, canonical filter name, etc.
  */
-struct FilterFacotryContext {
-  // Custom filter name from configuration.
-  absl::string_view custom_name;
-  // Canonical filter name.
-  absl::string_view filter_name;
+struct FilterContext {
+  // The name of the filter configuration that used to create related filter factory function.
+  // This could be any legitimate non-empty string.
+  std::string config_name;
+  // Canonical filter name. This is used to indicate type of filter. For example,
+  // "envoy.filters.http.buffer" for the HTTP buffer filter.
+  std::string filter_name;
```

Comment:
It's still necessary for now. Because this name will be used to search the per filter config, as a fallback of config name. 🤔 

If we deprecate the fallback policy and used the config name only someday, we can remove this name.

Reply:
I think the phrasing needs to be better. 
```
DEPRECATED: Filter extension qualified name. Used as a fallback for `config_name`.  E.g. ...
```

This is needed to support extension names in typed_config I think, so we can't change it to anything but what it was (which was extension qualified name).

Reply:
get it.